### ISSUE TITLE
Clarify implications of enabling hibernation

### ISSUE THREAD

**Issue by weiss** [blastrock/blastrock.github.io#10]
The [Kernel Lockdown feature][1] currently disables hibernation. The article gives the following explanation for why this is done:

> This is because during hibernation, you can boot another OS and edit the swap partition, replace kernel code and resume the system. This way you would get unsigned code into the kernel, which is not allowed by kernel lockdown, so it disables hibernation altogether.

It therefore concludes that this reasoning doesn't apply to the encrypted swap scenario:

> Except we don't care about this, since our swap is encrypted! Booting another OS will not allow anyone to inject code in the kernel.

The guide then mentions a kernel [patchset][2] that makes hibernation work under lockdown but hasn't been merged. And suggests a [kernel module][3] as an alternative, somewhat implying that security-wise, the patchset and the module yield a similar result given the "encrypted swap" scenario. (The guide _does_ mention that the module disables lockdown altogether rather than just enabling hibernation, but that's not my point.)

My point is: I think booting another OS to edit swap isn't the only attack vector lockdown attempts to protect against. Lockdown also tries to ensure an attacker with root access won't be able to tamper with protected kernel memory. If hibernation is allowed, that attacker could do just that by editing the unlocked swap data. The patchset you referenced goes to great lengths to address that scenario (see the author's [article on that topic][4] for the details).

Don't get me wrong: I'm not saying the kernel module shouldn't be used. As long as root has access to the key used for signing the UKI, there's no protection against root anyway. Users may well decide that they don't care about this attack vector. I'm just asking whether it wouldn't make sense to explicitly mention this trade-off.

TL;DR: I think it might be problematic to spread the _my swap is encrypted, so I'm safe!_ conclusion without mentioning the implications.

[1]: https://man7.org/linux/man-pages/man7/kernel_lockdown.7.html
[2]: https://lore.kernel.org/lkml/20210220013255.1083202-1-matthewgarrett@google.com/
[3]: https://github.com/blastrock/unlockdown
[4]: https://mjg59.dreamwidth.org/55845.html

**Comment by blastrock** [comment]
Hello, and thanks for the issue!

You are completely right, and I'm completely lazy, so sorry for the long delay. Also, it's almost 2025, and hibernation is not a feature on linux, and it saddens me.
